lumicks/pylake/kymotracker/detail/msd_estimation.py

```diff
@@ -447,3 +455,259 @@ def determine_optimal_points(frame_idx, coordinate, max_iterations=100):
+    """Expected variance of the diffusion estimate obtained with CVE.
+
+    The covariance-based diffusion estimator provides a simple unbiased estimator of diffusion.
+    This function is based on equation 24 from Vestergaard et al [2]_, but adapted for 1D.
+
+    Parameters
+    ----------
+    diffusion_constant : float
+        Estimate of the diffusion constant
+    variance_loc : float
+        Estimate of the localization variance
+    variance_variance_loc : float
+        Variance of the localization variance estimate.
+    dt : float
+        Time step
+    num_points : int
+        Number of points in the trace
+    blur_constant : float
+        Motion blur constant. See :func:`lumicks.pylake.kymotracker.detail.msd_estimation._cve` for
+        more information.
+    avg_frame_steps : float
+        Average frame steps. This number is the average of the time between two localizations in
+        frames mean(delta time between frame / frame timestep). If all frames had successful
+        localization, this constant will be 1 [2].
+
+    References
+    ----------
+    .. [1] Vestergaard, C. L., Blainey, P. C., & Flyvbjerg, H. (2014). Optimal estimation of
+           diffusion coefficients from single-particle trajectories. Physical Review E, 89(2),
+           022726.
+    .. [2] Vestergaard, C. L. (2016). Optimizing experimental parameters for tracking of diffusing
+           particles. Physical Review E, 94(2), 022401.
+    """
+    # Note that it uses a different definition of epsilon to circumvent a division by zero for D=0.
+    epsilon = variance_loc / dt - 2.0 * blur_constant * diffusion_constant
+    blur_term = (avg_frame_steps - 2.0 * blur_constant) ** 2
+    avg_diff = avg_frame_steps * diffusion_constant
+    numerator = 2.0 * avg_diff**2 + 4.0 * epsilon * avg_diff + 3.0 * epsilon**2
+    denominator = num_points * blur_term
+    term1 = numerator / denominator
+    term2 = variance_variance_loc / (blur_term * dt**2)
+    return term1 + term2
+
+
+def _cve(
+    frame_indices: npt.ArrayLike,
+    x: npt.ArrayLike,
+    dt: float,
+    blur_constant: float = 0,
+    variance_loc: Optional[float] = None,
+    variance_variance_loc: Optional[float] = None,
+) -> tuple:
+    """Covariance based estimator.
+
+    The covariance-based diffusion estimator provides a simple unbiased estimator of diffusion.
+    This estimator was introduced in the work of Vestergaard et al [1]. The correction for
+    missing data was introduced in [2].
+
+    Note that this estimator is only valid in the case of pure diffusion without drift.
+
+    References
+    ----------
+    .. [1] Vestergaard, C. L., Blainey, P. C., & Flyvbjerg, H. (2014). Optimal estimation of
+           diffusion coefficients from single-particle trajectories. Physical Review E, 89(2),
+           022726.
+    .. [2] Vestergaard, C. L. (2016). Optimizing experimental parameters for tracking of diffusing
+           particles. Physical Review E, 94(2), 022401.
```

Comment:
this should move to the bottom to be consistent with other docstrings. also watch out for the numbering, needs to be 4 & 5 so the docs would be formatted properly